packages/block-library/src/image/edit.js

```diff
@@ -333,7 +334,26 @@ export function ImageEdit( {
 	} );
 
 	// Much of this description is duplicated from MediaPlaceholder.
-	const isUrlAttributeConnected = !! metadata?.bindings?.url;
+	const { lockUrlControls = false } = useSelect(
```

Comment:
Technically speaking, there is no need to return an object if there is only a single return value. Anyway, we can refactor that later when we introduce the helper React hook as discussed in https://github.com/WordPress/gutenberg/issues/58233.